Missing steps in Dockerfile for installing ros-dependencies

nahsimai opened the pull request:
- Fixes missing steps for installing ros-dependencies. 

If there are some corrections/improvements to be made, I will be happy to help further.

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/carla-simulator/ros-bridge/632)
<!-- Reviewable:end -->

nahsimai committed:
fixes the issue about required ros-dependencies not being installed in docker image

BastOOn58 commented:
Since June 20th, 2023 the ROS Foxy Fitzroy distribution as reached its end-of-life (see [here](https://docs.ros.org/en/foxy/Releases.html)). As the ROS bridge works with this distribution, you must add the following argument to the rosdep update command to avoid any problem.
```
rosdep update --rosdistro=$ROS_DISTRO
```

nahsimai committed:
fixes to counter ROS Foxy EOL

Based on Comment from @BastOOn58: https://github.com/carla-simulator/ros-bridge/pull/632#issuecomment-1630629598

nahsimai commented:
> Since June 20th, 2023 the ROS Foxy Fitzroy distribution as reached its end-of-life (see [here](https://docs.ros.org/en/foxy/Releases.html)). As the ROS bridge works with this distribution, you must add the following argument to the rosdep update command to avoid any problem.
> 
> ```
> rosdep update --rosdistro=$ROS_DISTRO
> ```

Thanks for your comment @BastOOn58. I have made the relevant changes to the Dockerfile as suggested by you.
